python-connectors/pi-system_event-frames-search/connector.py

```diff
@@ -92,11 +95,16 @@ def generate_rows(self, dataset_schema=None, dataset_partitioning=None,
                             if not value:
```

Comment:
A comment on this whole section and possibily making it simpler:

If I understand correctly, in the response there is EITHER a "Value" key OR an "Items" key? And this is the case both for a non-batch response and a batch one, once the batch response is unpacked within get_rows_from_webids()? So once the batch response is unpacked it could be handled a bit like a non-batched one, maybe?

Looking at it now, I think both bits of code now handle the two possibilities (values or items) and the two sections lines 94-109 and lines 118 to 138 do roughly the same thing? These could probably be pulled out into a function and shared? (with a comment explaining the different structures it handles).

Reply:
was just a suggestion - deemed unneeded effort/risk